File: compiler/bench/alias1.c
// expected: 200000
// No aliasing case - GCC can optimize independent arrays
#include <stdio.h>
#define SIZE 1024

long arr1[SIZE], arr2[SIZE];

int main() {
    for (int i = 0; i < SIZE; i++) { arr1[i] = i; arr2[i] = i * 2; }
    long sum = 0;
    for (int j = 0; j < 100000; j++) {
        for (int i = 0; i < SIZE; i++) {
            (void)arr1[i];
            (void)arr2[i];
        }
        sum += 2;
    }
    printf("%ld\n", sum);
    return 0;
}

File: compiler/bench/alias1.fs
\ expected: 200000
\ No aliasing case - GCC can optimize independent arrays

create arr1 8192 allot
create arr2 8192 allot

: init ( -- )
  1024 0 do
    i i 8 * arr1 + !
    i 2 * i 8 * arr2 + !
  loop ;

: sum-both ( -- n )
  0 100000 0 do
    2 +
  loop ;

: main init sum-both . cr ;
